ingest/src/main/java/com/connexta/ingest/transform/TransformClient.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.ingest.transform;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.ResponseErrorHandler;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class TransformClient {
+
+  @Autowired private RestTemplate restTemplate;
+
+  private URI transformEndpoint;
+
+  public TransformResponse requestTransform(TransformRequest transformRequest) {
+    ResponseErrorHandler originalRequestHandler = restTemplate.getErrorHandler();
+    TransformResponse transformResponse = null;
+    try {
```

Comment:
Can you clarify what this syntax means? What happens if an exception gets thrown in the try? Does it get bubbled up somewhere?

Reply:
It turns that the default exception handler throws exception for reponse status codes outside of the 200 range. That isn't what we want to do because it would be hard to recover the JSON details/messages in the responses for malformed requests and server errors. I install a no-op error handler so I can extract an JSON the server sends in the response.

I let Spring the RestTemplate object, which is shared with other classes in the environment  so I don't want to replace the default error handler for everyone. I temporarily replace the error handler. 

There is probably a more elegant way to do this.